_posts/2020-02-20-ddl.md

```diff
@@ -0,0 +1,124 @@
+
+Users can define tables over static data sets, anything from a local CSV file to a full-fledged data lake or even Hive.
+Leveraging Flink's efficient batch processing capabilities, they can perform ad-hoc queries searching for exciting insights.
+Once something interesting is identified, businesses can gain real-time and continuous insights by merely altering the table so that it is powered by a message queue such as Kafka.
+Because Flink guarantees SQL queries have unified semantics over batch and streaming, users can be confident that redeploying this query as a continuous streaming application over a message queue will output identical results.
+
+{% highlight sql %}
+-- Define a table called orders that is backed by a Kafka topic
+-- The definition includes all relevant Kafka properties,
+-- the underlying format (JSON) and even defines a
+-- watermarking algorithm based on one of the fields
+-- so that this table can be used with event time.
+CREATE TABLE orders (
+	user_id    BIGINT,
+	product    STRING,
+	order_time TIMESTAMP(3),
+	WATERMARK FOR order_time AS order_time - '5' SECONDS
+) WITH (
+	'connector.type'    	 = 'kafka',
+	'connector.version' 	 = 'universal',
+	'connector.topic'   	 = 'orders',
+	'connector.startup-mode' = 'earliest-offset',
+	'connector.properties.bootstrap.servers' = 'localhost:9092',
+	'format.type' = 'json' 
+);
+
+-- Define a table called product_analsysis
+-- on top of ElasticSearch 7 where  we 
+-- can write the results of our query. 
+CREATE TABLE product_analysis (
+	product 	STRING,
+	tracking_time 	TIMESTAMP(3),
+	units_sold 	BIGINT
+) WITH (
+	'connector.type'    = 'elasticsearch',
+	'connector.version' = '7',
+	'connector.hosts'   = 'http://host_name:9092',
+	'connector.index'   = 'ProductAnalysis',
+	'connector.document.type' = 'analysis' 
+);
+
+-- A simple query that analyzes order data
+-- from Kafka and writes results into 
+-- ElasticSearch. 
+INSERT INTO product_analysis
+SELECT
+	product_id,
+	TUMBLE_START(order_time, INTERVAL '1' DAY) as tracking_time,
+	COUNT(*) as units_sold
+FROM orders
+GROUP BY
+	product_id,
+	TUMBLE(order_time, INTERVAL '1' DAY);
+{% endhighlight %}
+
+# Catalogs
+
+While being able to create tables is important, it often isn't enough.
+A business analyst, for example, shouldn't have to know what properties to set for Kafka, or even have to know what the underlying data source is, to be able to write a query.
+
+To solve this problem, Flink 1.10 also ships with a revamped catalog system for managing metadata about tables.
+With catalogs, users can create tables once and reuse them across Jobs and Sessions.
+Now, the team managing a data set can create a table and immediately make it accessible to other groups within their organization.
+
+The most notable catalog that Flink integrates with today is Hive Metastore.
+The Hive catalog allows Flink to fully interoperate with Hive and serve as a more efficient query engine.
```

Comment:
Be more explicit here: full integration means reading from and writing to Hive tables and using Hive UDFs?